ngraph/test/runtime/ie/unit_test.manifest

```diff
@@ -1139,6 +1139,17 @@ IE_CPU.log_softmax_2d_axis_neg2
 IE_CPU.log_softmax_3d_axis_0
 IE_CPU.log_softmax_3d_axis_neg3
 
+# Unsupported NMS-5
+IE_CPU.nonmaxsuppression_center_point_box_format
```

Comment:
CPU supports NMS-3. So I assume that the tests must be supported. Why they are disabled?

Reply:
The reason is the following: all previous NMS are converted to NMS-5, but CPU doesn't support NMS-5 yet.

Reply:
I think that some of these tests should be enabled because NMS-4 which was supported on CPU is a sub-set of NMS-5.

Reply:
When I commented all these tests in `ngraph/test/runtime/ie/unit_test.manifest`, I got
```sh
[ RUN      ] IE_CPU.nonmaxsuppression_center_point_box_format
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 9 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_center_point_box_format (5 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_flipped_coordinates
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 9 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_flipped_coordinates (6 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_identical_boxes
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 9 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_identical_boxes (6 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_limit_output_size
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 6 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_limit_output_size (5 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_single_box
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 3 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_single_box (6 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_suppress_by_IOU
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 9 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_suppress_by_IOU (5 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_suppress_by_IOU_and_scores
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 9 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_suppress_by_IOU_and_scores (6 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_two_batches
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 12 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_two_batches (6 ms)
[ RUN      ] IE_CPU.nonmaxsuppression_two_classes
unknown file: Failure
C++ exception with description "The output blob size is not equal to the network output size: got 12 expecting 1" thrown in the test body.
[  FAILED  ] IE_CPU.nonmaxsuppression_two_classes (5 ms)
```
These exceptions are raised in `inference-engine/src/cldnn_engine/cldnn_infer_request.cpp`.

Reply:
But MKLDNN has tests (`./inference-engine/tests_deprecated/unit/engines/mkldnn/graph/layers/extensions/non_max_suppression_tests.cpp`) with input data and output data as in tests `non_max_suppression*` from the file `ngraph/test/backend/non_max_suppression.in.cpp`, and these MKLDNN tests are passed

Reply:
@iefode , could you comment on this issue? Why enabling CPU tests results in failures on GPU?

Reply:
I'm sorry, this exception can be raised from `inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_request_internal.hpp` (line 343) also. The reason of the exception raising  is the following. The nGraph function for NMS-5 are passed to the method `compile()` of the backend, but **this method doesn't apply any nGraph transformations to the passed function**, and the plugin backend gets `CNNNetwork` with NMS-5, NMS-5 has dynamic shapes for two of three outputs, and results of these two outputs are interpreted as scalars.

If we apply all needed nGraph transformations to the nGraph function with NMS-5 to get the nGraph function with NMSIE3 (internal operation, similar with NMS-5, but with all static output shapes), before the method `compile()` call, then tests for INTERPRETER backend for NMS-5 will be failed, because NMSIE3 has not the reference implementation, but NMS-5 has one.

Hence, we cannot enable tests `IE_CPU.nonmaxsuppression*`.